lib/eclipse/include/opm/parser/eclipse/EclipseState/Schedule/MSW/SpiralICD.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+  Copyright 2017 SINTEF Digital, Mathematics and Cybernetics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SPIRALICD_HPP_HEADER_INCLUDED
+#define SPIRALICD_HPP_HEADER_INCLUDED
+
+#include <opm/parser/eclipse/Deck/DeckRecord.hpp>
+#include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
+
+#include <limits>
+#include <map>
+#include <utility>
+
+
+namespace Opm {
+
+struct SpiralICD {
```

Comment:
Now - this was a real pattern breaker:
1. Implementation in `.hpp` file.
2. `struct` with public members instead of a class with private members.
3. No test?! New (non trivial) classes shall have tests!

I am not saying the patterns are sacred, but I would prefer you follow them unless there is a specific reason not to.

Reply:
> Please confirm a .cpp file for implementation is what you want here.

yes - that is what I want.

Reply:
> I can change to a class

Yes please; preferably with private members - but I will not insist.

Reply:
I just provided some fixes to your comments. Please let me know if they are what you want.